### PR TITLE
Managed pastures tau

### PR DESCRIPTION
**Pull request by mppalves**
## :bird: Purpose of this PR :bird:

- Managed pastures implementation PR 1: Added a new tau dimension for managed pastures.

## :wrench: Checklist for PR creator :wrench:

- [x] Labeling pull request correctly [from the label list](https://github.com/magpiemodel/magpie/labels).

* Low risk : Simple bugfixes (missing files, updated documentation, typos) or Start/output scripts
* Medium risk : New realization / Changes to existing realization / Other changes which don't modify default.cfg
* High risk : New input files (if cfg$input is changed in default.cfg) / Modification to core model (eg. changes in equations, calculations, introduction of new sets etc.) / Other changes in default.cfg

- [x] Providing additional information based on PR label

* Low risk : No new model run needed.
* Medium risk : Default run based on the current version of the fork from which PR is made
* High risk
  * Default run from the current develop branch: /p/projects/landuse/users/pedrosa/MAgPIE/develop_clean/output/default_2022-01-30_17.04.22
  * Default run based on the current version of the fork from which PR is made: /p/projects/landuse/users/pedrosa/MAgPIE/MAgPIE_clean/output/default_2022-01-30_16.11.39

- [x] :chart_with_downwards_trend: Performance loss/gain from current default behavior :chart_with_upwards_trend:
  * Current develop branch default :  32 mins
  * This PR's default :  33 min
 
- [x] Added changes to `CHANGELOG.md`
- [x] Compilation check (model starts without compilation errors - use `gams main.gms action=c` in model folder for testing).
- [x] No hard coded numbers and cluster/country/region names.
- [x] The new code doesn't contain declared but unused parameters or variables.
- [x] Where relevant, In-code comments added including documentation comments.
- [x] Made sure that documentation created with [`goxygen`](https://github.com/pik-piam/goxygen) is okay (use `goxygen::goxygen()` for testing).
- [x] Changes to [`magpie4`](https://github.com/pik-piam/magpie4) R library for post processing of model output (ideally backward compatible).
- [x] Self-review of my own code.
- [x] For high risk runs: validation of major model indicators - Land-use, emissions, food prices, Tau.  %Delete this line in case it is not a high risk run%

## :warning: Additional notes or warnings :warning:
NA

## :rotating_light: Checklist for RSE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] No unnecessary increase in module interfaces
- [ ] All required updates in interfaces (if any) have been properly adressed in the module contracts
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

## :rotating_light: Checklist for MAgPIE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] Changes to the model are scientifically sound
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

### CODE REVIEW

File: modules/13_tc/endo_jun18/declarations.gms [hunk 1]
```diff
@@ -6,30 +6,30 @@
...
+ q13_cost_tc(i, tautype)              Costs for TC (mio. USD05PPP per yr)
 ;
 
 parameters
- pc13_land(i)                Crop land area per region (mio ha)
- pcm_tau(h)                  Tau factor of the previous time step (1)
- pc13_tcguess(h)             Guess for annual tc rates in the next time step (1)
- i13_tc_factor(t)            Regression factor (USD05PPP per ha)
- i13_tc_exponent(t)          Regression exponent (1)
+ pc13_land(i, tautype)                Crop land area per region (mio ha)
```
Comment (emolinab) [inline modules/13_tc/endo_jun18/declarations.gms:20]: Description change, Now it includes pasture as well

File: modules/10_land/feb15/declarations.gms [hunk 2]
```diff
@@ -9,6 +9,7 @@
 parameters
  pm_land_start(j,land)            Land initialization area (mio. ha)
  pcm_land(j,land)                 Land area in previous time step (mio. ha)
+ pcm_grass(j,grassland)           Grassland areas in previous time step (mio. ha)
```
Comment (emolinab) [inline modules/10_land/feb15/declarations.gms:12]: pcm_grass is only defined in 1995 and I didn't see any other update of this parameter during the runs. It only feeds pc13_land(i,"pastr"). So maybe change the description/name to specify that it is1995/start grassland areas?

**Review by emolinab — CHANGES_REQUESTED** [review]

File: modules/10_land/feb15/declarations.gms (hunk 2 above)
Comment (emolinab) [reply modules/10_land/feb15/declarations.gms:12]: ? Is this going to be used later?

**Review by flohump — COMMENTED** [review]
Great to see this development moving!
Just for clarification. With this PR you add data sets and dimensions for managed pasture. But these additions are not yet used, right?
For instance, pasture is not yet split and pasture yields are not yet changed. Is that correct?
If I'm not mistaken vm_tau(i,"pastr") is currently a free variable.

**Comment by mppalves** [comment]
> Great to see this development moving! Just for clarification. With this PR you add data sets and dimensions for managed pasture. But these additions are not yet used, right? For instance, pasture is not yet split and pasture yields are not yet changed. Is that correct? If I'm not mistaken vm_tau(i,"pastr") is currently a free variable.

Hi Florian! Yes, I am splitting the whole implementation into chunks. This is the first chunk of code, so vm_tau(i, "past") is still not engaged in any other module. It will be once all the other parts are put together!

File: modules/13_tc/endo_jun18/declarations.gms [hunk 3]
```diff
@@ -6,30 +6,30 @@
 *** |  Contact: magpie@pik-potsdam.de
 
 positive variable
- vm_tau(h)                   Agricultural land use intensity tau (1)
- vm_tech_cost(i)             Annuitized costs of TC (mio. USD05PPP per yr)
- v13_cost_tc(i)              Technical change costs per region (mio. USD05PPP)
+ vm_tau(h,tautype)                   Agricultural land use intensity tau (1)
+ vm_tech_cost(i,tautype)             Annuitized costs of TC (mio. USD05PPP per yr)
+ v13_cost_tc(i,tautype)              Technical change costs per region (mio. USD05PPP)
 ;
```
Comment (tscheypidi) [inline modules/13_tc/endo_jun18/declarations.gms:12]: your implementation is completely different to what is there so far. Hence, you need to create a new realization and do your changes in this new realization.

Comment (tscheypidi) [reply modules/13_tc/endo_jun18/declarations.gms:12]: and try to reduce your changes in the interfaces! Changing interfaces will introduce a lot of work, e.g. a lot of changes in the `magpie4` package and so on

File: modules/10_land/feb15/declarations.gms (hunk 2 above)
Comment (tscheypidi) [inline modules/10_land/feb15/declarations.gms:12]: I don't think that you need that additional interface...in the tc module you could use `pcm_land("past")` instead even so it has a slightly different meaning

**Review by tscheypidi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**